Assets/FPS/Scripts/Gameplay/Nautica/BotKillerAgent.cs

```diff
@@ -69,187 +70,203 @@ public override void OnEpisodeBegin()
 			enemies = trainingLevelManager.enemies;
 			pickups = trainingLevelManager.pickups;
 
-            trackedHealth = 1f;
+			trackedHealth = 1f;
 			trackedEnemyHealth.Clear();
 			foreach (var e in enemies) trackedEnemyHealth.Add(1f);
+		}
+
+		private void DebugLogReward(float reward, string rewardtype)
+		{
+			string rewardOutput = "<b>Reward:</b> ";
+			if (reward < 0) rewardOutput = "<color=red>";
+			if (reward > 0) rewardOutput = "<color=green>";
+			rewardOutput += rewardtype + " <b>" + reward.ToString() + "</b></color>, STEP: " + StepCount.ToString() + " Cumulative: " + GetCumulativeReward().ToString();
+			Debug.unityLogger.Log(LOGTAG, rewardOutput);
+		}
+
+		public override void CollectObservations(VectorSensor sensor)
+		{
+			ObserveHealth(sensor);
+			ObserveAmmo(sensor);
+			ObserveEnemies(sensor);
+			ObservePickups(sensor);
+		}
+
+		private void ObserveHealth(VectorSensor sensor)
+		{
+			var healthComponent = GetComponent<Unity.FPS.Game.Health>();
+			float normalizedHealth = healthComponent.GetRatio();
+			sensor.AddObservation(normalizedHealth);
+			UpdateTrackedAgentHealthAndRewardAgent(normalizedHealth);
+		}
+
+		private void UpdateTrackedAgentHealthAndRewardAgent(float agentHealth)
```

Comment:
normally we'd like to have a method do 1 task, this is doing 2 (update and reward)
but at the same time, the way we handle observations and rewards, it's best to keep them together so it makes sense

Reply:
split into different methods UpdateAgentTrackedHealth() and DisplayHealthReward() 🤠